chirp/drivers/uvk5.py

```diff
@@ -0,0 +1,1972 @@
+
+        chirp_common.split_tone_decode(mem, (tx_tmode, tx_tone, tx_pol),
+                                       (rx_tmode, rx_tone, rx_pol))
+
+    # Extract a high-level memory object from the low-level memory map
+    # This is called to populate a memory in the UI
+    def get_memory(self, number2):
+        number = number2-1  # in the radio memories start with 0
+
+        mem = chirp_common.Memory()
+
+        # cutting and pasting configs from different radios
+        # might try to set channel 0
+        if number2 == 0:
+            LOG.warning("Attempt to get channel 0")
+            return mem
+
+        _mem = self._memobj.channel[number]
+
+        tmpcomment = ""
+
+        mem.number = number2
+
+        is_empty = False
+        # We'll consider any blank (i.e. 0MHz frequency) to be empty
+        if (_mem.freq == 0xffffffff) or (_mem.freq == 0):
+            is_empty = True
+
+        tmpscn = SCANLIST_LIST[0]
+
+        # We'll also look at the channel attributes if a memory has them
+        if number < 200:
+            _mem3 = self._memobj.channel_attributes[number]
+            # free memory bit
+            if _mem3.is_free > 0:
+                is_empty = True
+            # scanlists
+            if _mem3.is_scanlist1 > 0 and _mem3.is_scanlist2 > 0:
+                tmpscn = SCANLIST_LIST[3]  # "1+2"
+            elif _mem3.is_scanlist1 > 0:
+                tmpscn = SCANLIST_LIST[1]  # "1"
+            elif _mem3.is_scanlist2 > 0:
+                tmpscn = SCANLIST_LIST[2]  # "2"
+
+        if is_empty:
+            mem.empty = True
+            # set some sane defaults:
+            mem.power = UVK5_POWER_LEVELS[2]
+            mem.extra = RadioSettingGroup("Extra", "extra")
+            rs = RadioSetting("bclo", "BCLO", RadioSettingValueBoolean(False))
+            mem.extra.append(rs)
+            rs = RadioSetting("frev", "FreqRev",
+                              RadioSettingValueBoolean(False))
+            mem.extra.append(rs)
+            rs = RadioSetting("pttid", "PTTID", RadioSettingValueList(
+                PTTID_LIST, PTTID_LIST[0]))
+            mem.extra.append(rs)
+            rs = RadioSetting("dtmfdecode", "DTMF decode",
+                              RadioSettingValueBoolean(False))
+            mem.extra.append(rs)
+            rs = RadioSetting("scrambler", "Scrambler", RadioSettingValueList(
+                SCRAMBLER_LIST, SCRAMBLER_LIST[0]))
+            mem.extra.append(rs)
+
+            rs = RadioSetting("scanlists", "Scanlists", RadioSettingValueList(
+                SCANLIST_LIST, SCANLIST_LIST[0]))
+            mem.extra.append(rs)
+
+            # actually the step and duplex are overwritten by chirp based on
+            # bandplan. they are here to document sane defaults for IARU r1
+            # mem.tuning_step = 25.0
+            # mem.duplex = "off"
+
+            return mem
+
+        if number > 199:
+            mem.name = VFO_CHANNEL_NAMES[number-200]
```

Comment:
Didn't see the extd_number functionality. Looking into it.

Reply:
Changed to use extd_number, however still not sure what to do with the name field?
- leave it blank and immutable
- put a description in it and make it immutable
- put a description in it and make it immutable, change extd_number to VFO-n (where n=1..14)
 
The first option is bad, because the user doesn't see all of the label, and there is no way to make the memory number field wider.

The second option is better because the description in the channel number doesn't fit. And i'd like to use the names that the vendor software uses, for example F4(174M-350M)A

The third option is best, because it informs the user that these are VFOs, and also shows the named from the vendor software in the name field.

Pick any version you want, and i will do it (but please try to pick the better-looking and more understandable for the user option).

Reply:
> Changed to use extd_number, however still not sure what to do with the name field?
> 
>     * leave it blank and immutable
> 
>     * put a description in it and make it immutable
> 
>     * put a description in it and make it immutable, change extd_number to VFO-n (where n=1..14)
> 
> 
> The first option is bad, because the user doesn't see all of the label, and there is no way to make the memory number field wider.
> 
> The second option is better because the description in the channel number doesn't fit. And i'd like to use the names that the vendor software uses, for example F4(174M-350M)A
> 
> The third option is best, because it informs the user that these are VFOs, and also shows the named from the vendor software in the name field.
> 
> Pick any version you want, and i will do it (but please try to pick the better-looking and more understandable for the user option).

The first option is the only viable one. Not only because you do *not* own the name field and cannot use it for random things, but also because that's the behavior of the other several hundred drivers here. Consistency wins. Please have some respect for the fact that this is just one driver of many hundreds currently supported in the code base.

I can work on resolving the issue with the display in some way, but in the mean time you should make the extd_number small enough that it fits. If you just drop the "-ENDFREQ" part of the name it should fit fine.